### PR TITLE
MySQL and PostgreSQL CLI update with new business model.

### PR DESCRIPTION
**Pull request by rohit-joy**
MySQL and PostgreSQL CLI update with new business model.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [x] Each command and parameter has a meaningful description.
- [x] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

**Comment by promptws** [comment]
View a preview at https://prompt.ws/r/Azure/azure-cli/5685
This is an **experimental preview** for @microsoft users.

**Comment by rohit-joy** [comment]
@QingqingYuan, @derekbekoe, please review.

**Comment by rohit-joy** [comment]
@tjprescott, Please review.

File: src/command_modules/azure-cli-rdbms/HISTORY.rst [hunk 1]
```diff
@@ -3,6 +3,10 @@
 Release History
 ===============
 
+0.0.13
+++++++
+* Preview release with new business model.
```
Comment (tjprescott) [inline src/command_modules/azure-cli-rdbms/HISTORY.rst]: This should be more specific.  What commands/parameters changed?

File: src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/_params.py [hunk 2]
```diff
@@ -51,28 +45,35 @@ def _complex_params(command_group, engine):
...
         with self.argument_context(scope) as c:
+            c.ignore('size', 'family', 'capacity', 'tier')
+
             c.argument('server_name', options_list=['--name', '-n'], id_part='name', help='Name of the server.')
             c.argument('administrator_login', options_list=['--admin-user', '-u'])
             c.argument('administrator_login_password', options_list=['--admin-password', '-p'], required=False, help='The password of the administrator login.')
             c.argument('ssl_enforcement', arg_type=get_enum_type(['Enabled', 'Disabled']), options_list=['--ssl-enforcement'], help='Enable ssl enforcement or not when connect to server.')
-            c.argument('tier', arg_type=get_enum_type(['Basic', 'Standard']), options_list=['--performance-tier'], help='The performance tier of the server.')
-            c.argument('capacity', options_list=['--compute-units'], type=int, help='Number of compute units.')
+            c.argument('tier', arg_type=get_enum_type(['Basic', 'GeneralPurpose', 'MemoryOptimized']), required=False, options_list=['--performance-tier'], help='The performance tier of the server.')
```
Comment (tjprescott) [inline src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/_params.py]: Is there no enum in your SDK for this?  Also, why do you have the kwarg `required=False` so many times?  In the overwhelming majority of cases, it shouldn't be needed.

File: src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/_params.py [hunk 3]
```diff
@@ -51,28 +45,35 @@ def _complex_params(command_group, engine):
...
+            c.argument('capacity', options_list=['--vcore'], type=int, required=False, help='Number of vcore.')
+            c.argument('family', options_list=['--family'], arg_type=get_enum_type(['Gen4', 'Gen5']), required=False, help='Hardware generation.')
             c.argument('storage_mb', options_list=['--storage-size'], type=int, help='The max storage size of the server, unit is MB.')
+            c.argument('backup_retention_days', options_list=['--backup-retention'], type=int, help='The max days of retention, unit is days.')
```
Comment (tjprescott) [inline src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/_params.py]: "The max days of retention, unit is days."  This is redundant (and not a proper sentence).  Simply "The number of days a backup is retained." would suffice.

File: src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/_params.py [hunk 4]
```diff
@@ -51,28 +45,35 @@ def _complex_params(command_group, engine):
...
             c.argument('tags', tags_type)
 
     for scope in ['mysql server-logs', 'postgres server-logs']:
         with self.argument_context(scope) as c:
-            c.argument('file_name', options_list=['--name', '-n'], nargs='+', help='Space-separated list of log filenames on the server to download.')
+            c.argument('file_name', options_list=['--name', '-n'], nargs='+', help='Space separated list of log filenames on the server to download.')
```
Comment (tjprescott) [inline src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/_params.py]: This should say "Space-separated"

File: src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/commands.py [hunk 5]
```diff
@@ -17,85 +25,107 @@ def load_command_table(self, _):
 
     mysql_servers_sdk = CliCommandType(
         operations_tmpl='azure.mgmt.rdbms.mysql.operations.servers_operations#ServersOperations.{}',
+        client_arg_name='self',
```
Comment (tjprescott) [inline src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/commands.py]: Please remove ALL references to client_arg_name.  It is no longer needed for extensions and it was never needed for modules.

File: src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/commands.py [hunk 6]
```diff
@@ -17,85 +25,107 @@ def load_command_table(self, _):
...
         operations_tmpl='azure.mgmt.rdbms.postgresql.operations.databases_operations#DatabasesOperations.{}',
+        client_arg_name='self',
         client_factory=cf_postgres_db
     )
 
     with self.command_group('mysql server', mysql_servers_sdk, client_factory=cf_mysql_servers) as g:
-        g.command('create', 'create_or_update')
-        g.custom_command('restore', '_server_restore')
+        g.command('create', 'create')
+        g.custom_command('restore', '_server_restore', no_wait_param='no_wait')
```
Comment (tjprescott) [inline src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/commands.py]: We have recently merged a change to how --no-wait is handled.  Please see: https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md and search for the section "Supporting --no-wait"

File: src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/tests/test_rdbms_commands.py [hunk 7]
```diff
@@ -77,19 +81,21 @@ def _test_server_mgmt(self, database_engine, resource_group_1, resource_group_2)
...
+        new_cu = 4
+        family = 'Gen4'
+        skuname = '{}_{}_{}'.format("GP", family, old_cu)
 
         rg = resource_group_1
-        loc = 'westeurope'
+        loc = 'brazilsouth'
 
         # test create server
         self.cmd('{} server create -g {} --name {} -l {} '
```
Comment (tjprescott) [inline src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/tests/test_rdbms_commands.py]: Highly recommended to use the kwarg style syntax for tests, for readability.  https://github.com/Azure/azure-cli/blob/dev/doc/authoring_tests.md#sample-5-get-more-from-resourcegrouppreparer

File: src/command_modules/azure-cli-rdbms/HISTORY.rst [hunk 8]
```diff
@@ -3,6 +3,10 @@
 Release History
 ===============
 
+0.0.13
```
Comment (tjprescott) [inline src/command_modules/azure-cli-rdbms/HISTORY.rst]: Does this have breaking changes relative to the old model?  if so, you should bump the minor version.

File: src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/commands.py (hunk 5 above)
Comment (rohit-joy) [reply src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/commands.py]: Done.

File: src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/_params.py (hunk 4 above)
Comment (rohit-joy) [reply src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/_params.py]: Done.

File: src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/_params.py (hunk 3 above)
Comment (rohit-joy) [reply src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/_params.py]: Done.

File: src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/_params.py (hunk 2 above)
Comment (rohit-joy) [reply src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/_params.py]: Done

File: src/command_modules/azure-cli-rdbms/HISTORY.rst (hunk 8 above)
Comment (rohit-joy) [reply src/command_modules/azure-cli-rdbms/HISTORY.rst]: Yes, this one will run against new API.

File: src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/commands.py (hunk 6 above)
Comment (rohit-joy) [reply src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/commands.py]: Done.

File: src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/tests/test_rdbms_commands.py (hunk 7 above)
Comment (tjprescott) [reply src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/tests/test_rdbms_commands.py]: The doc has been updated based on your feedback.

**Pull request merged**